src/lxc/cmd/lxc_usernsexec.c

```diff
@@ -224,37 +225,42 @@ static int read_default_map(char *fnam, int which, char *user)
 		if (!p2)
 			continue;
 
-		newmap = malloc(sizeof(*newmap));
-		if (!newmap)
-			goto on_error;
+		line[strlen(line) - 1] = '\0';
+		*p2 = '\0';
 
-		ret = lxc_safe_ulong(p1 + 1, &newmap->hostid);
+		ret = lxc_safe_ulong(p1 + 1, &ul1);
 		if (ret < 0)
-			goto on_error;
+			break;
 
-		ret = lxc_safe_ulong(p2 + 1, &newmap->range);
+		ret = lxc_safe_ulong(p2 + 1, &ul2);
 		if (ret < 0)
-			goto on_error;
+			break;
+
+		ret = -1;
+		newmap = malloc(sizeof(*newmap));
+		if (!newmap)
+			break;
 
 		newmap->nsid = 0;
 		newmap->idtype = which;
+		newmap->hostid = ul1;
+		newmap->range = ul2;
 
-		ret = -1;
 		tmp = malloc(sizeof(*tmp));
-		if (!tmp)
-			goto on_error;
+		if (!tmp) {
+			free(newmap);
+			break;
+		}
 
```

Comment:
This is ok, but another fix would have been to set

```
newmap = NULL;
```

here.

But actually, what you have here changes behavior.  And I think fixes another bug.

If there are no matching entries, you now return -1.  Previously it returned 0 in that case, which the caller took to mean "there was a matching entry".